### PR TITLE
Bug 1371442 - Initialize UAOverrides after UserAgentsOverride.jsm

### PR DESCRIPTION
**Pull request by denschub**
r? @MDTsai

### CODE REVIEW

File: src/bootstrap.js [hunk 1]
```diff
@@ -60,8 +60,15 @@ this.startup = function({webExtension}) {
   Services.prefs.clearUserPref(UA_ENABLE_PREF_NAME);
   Services.prefs.addObserver(UA_ENABLE_PREF_NAME, UAEnablePrefObserver, false);
 
-  overrider = new UAOverrider(UAOverrides);
-  overrider.init();
+  let messageManager = Components.classes["@mozilla.org/globalmessagemanager;1"]
```
Comment (MDTsai) [inline src/bootstrap.js]: nit: to make it short?
const { classes: Cc } = Components;
let messageManager = Cc["@mozilla.org/globalmessagemanager;1"]
                                            .getService(Ci.nsIMessageListenerManager);
 Besides, add comment with bug ID to explain reason for this. Then that will be good to merge :)

Comment (denschub) [inline src/bootstrap.js]: I usually avoid defining `Components.*` shorthands if they only get used once, but you are right, it visual makes a difference here. :)

**Comment by denschub** [comment]
@MDTsai fixed your remarks, interdiff here:

```diff
diff --git a/src/bootstrap.js b/src/bootstrap.js
index 05e3819..b883252 100644
--- a/src/bootstrap.js
+++ b/src/bootstrap.js
@@ -2,8 +2,10 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
-Components.utils.import("resource://gre/modules/Services.jsm");
+const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
 
 const PREF_BRANCH = "extensions.webcompat.";
 const PREF_DEFAULTS = {perform_ua_overrides: true};
@@ -60,8 +62,11 @@ this.startup = function({webExtension}) {
   Services.prefs.clearUserPref(UA_ENABLE_PREF_NAME);
   Services.prefs.addObserver(UA_ENABLE_PREF_NAME, UAEnablePrefObserver, false);
 
-  let messageManager = Components.classes["@mozilla.org/globalmessagemanager;1"]
-    .getService(Components.interfaces.nsIMessageListenerManager);
+  // Listen to the first Browser:FirstPaint message we get and initialize our
+  // overrider there. This is done to avoid slowing down the apparent startup
+  // proces, since we avoid loading anything before the first window is visible
+  // to the user. See bug 1371442 for details.
+  let messageManager = Cc["@mozilla.org/globalmessagemanager;1"].getService(Ci.nsIMessageListenerManager);
   let onFirstPaint = () => {
     messageManager.removeMessageListener("Browser:FirstPaint", onFirstPaint);
     overrider = new UAOverrider(UAOverrides);

```

**Comment by denschub** [comment]
Okay, will have to work on this again. Mike Conley suggested using the [`browser-delayed-startup-finishe`](https://developer.mozilla.org/en-US/docs/Observer_Notifications#Browser) notification instead of listening to `Browser:FirstPaint`, so we should do that instead. Will update this PR!

**Comment by denschub** [comment]
Sorry for the push notification - I pushed changes based on another bug I am waiting for review on. I'll ask for review as soon as the other thing lands.

**Comment by denschub** [comment]
New notification has landed.

r? @MDTsai

**Comment by denschub** [comment]
gnah... reopening, since I accidentally closed it by renaming the branch and pushing, this has not been merged yet.

**Pull request merged**